### DISCUSSION TITLE
How to verify it's working?

### DISCUSSION THREAD

**Discussion by artemislena** [struct/isoalloc#229] [Q&A]
T.: So, we're on macOS (ARM). We downloaded the latest release source, built w `make cpp_library` (that should work for both C and C++ programs then, right?), copied the `.dylib` file from `./build` into `/usr/local/lib`, n put the path for that into `DYLD_INSERT_LIBRARIES` in `~/.zshenv` (i.e. `DYLD_INSERT_LIBRARIES=/usr/local/lib/libisoalloc.dylib`). So far no errors or anything, but is't actually being preloaded n used? N do we need the `.o` files (also in `./build`) for anything? And's there any way for making graphic apps (outside the terminal) use't as an allocator, too?

**Comment by struct** [comment]
I missed this notification, I will take a look later this week.

**Comment by struct** [comment]
Hooking code on MacOS is notoriously tricky and new versions of MacOS often break what was previously working. Here are some ideas to confirm the library is loaded:
* Try turning off SIP
* Use `DYLD_PRINT_LIBRARIES=1 DYLD_FORCE_FLAT_NAMESPACE=1` to see if the dylib is loaded
* Build a debug version of the library `make cpp_library_debug` and see if it prints status in the destructor.

If these show its not working let me know. However it is best to use isoalloc through the APIs directly.

**Reply by artemislena** [reply]
L: So uh, we'd rather not disable SIP (and don't know how to anyway), the second point seems to do nothing (at least without the first), and uh, as for the third, we don't know what printing a status in a destructor means.
